### DISCUSSION TITLE
Laravel package route not found

### DISCUSSION THREAD

**Discussion by PrasadChinwal** [laravel/framework#47064] [Q&A]
I am building a package for Laravel v10.9 and have loaded routes in my Service Provider boot method like so:
```
$this->loadRoutesFrom(__DIR__ . '/../routes/routes.php');
```
When I run `php artisan route:list`, I can see all my routes.
| Type| route | name | Action|
|----| ----- | ----- | ----- |
|GET HEAD |   auth/callback .................................. |shib.callback |  PrasadChinwal\Shibboleth\Actions\AuthHandler@login |
|GETHEAD  |  login ....................................................................................................... |login | |
|GETHEAD  |  logout ............................................... | logout | PrasadChinwal\Shibboleth\Actions\AuthHandler@logout |

Unfortunately when i try to visit the login route I get 404. I have tried:
- move the `loadRoutesFrom` to the register method instead of boot.
- `composer dump-autoload`
- `php artisan cache:clear` `php artisan route:clear` `php artisan config:clear`
- Tried to manually register my Service provider in the `config/app.php` file after and before Laravel's `RouteServiceProvider`.

Let me know if more details are needed.

**Comment by riomigal** [comment]
Did you find a solution for this issue? I am facing the same problem, it's very weird since it was working before without caching. Now I do need to cache the routes to make it work...

**Reply by PrasadChinwal** [reply]
Unfortunately no. I am still facing same issues. Are you saying that if you cache the routes it will work?

**Comment by riomigal** [comment]
Yes, caching worked for me, but it's not the solution I want. I will have a
closer look the next days

**Reply by PrasadChinwal** [reply]
Please let me know if you find a solution.

**Comment by riomigal** [comment]
That’s what worked for me. If you are loading the package from a custom repository and you do have a sys link.

1. Delete your package in the vendor folder and run again composer install. This should install the package again in the vendor directory.

After this everything worked again as it should for me… (Even without caching)

**Comment by adityadees** [comment]
what I'm do when I facing a problem with the laravel cache when create package is
- clear cache (depend of your cache type), ex: if you do a config so you must do clear cache for config, if route so clear cache for route
- remove the package from vendor, you can use `composer remove package_name` but if stil can't you can remove manual, just remove your package folder inside vendor folder like @riomigal  said above
- remove `composer.lock`
- clear composer cache just run `composer clearcache`
- last try re-install your package
